app/src/main/java/io/neurolab/communication/bluetooth/BluetoothConnectionManager.java

```diff
@@ -0,0 +1,141 @@
+        if(bluetoothConnectionManager == null){
+            bluetoothConnectionManager = new BluetoothConnectionManager();
+        }
+        return bluetoothConnectionManager;
+    }
+
+    /*
+    This method checks the current bluetooth state and returns any one of the three states
+    NO_BLUETOOTH_SUPPORT, BLUETOOTH_ENABLED, BLUETOOTH_DISABLED
+     */
+    public int checkBluetoothState(){
+        if(bluetoothAdapter == null)
+            return NO_BLUETOOTH_SUPPORT;
+        else if(!bluetoothAdapter.isEnabled())
+            return BLUETOOTH_DISABLED;
+        else
+            return BLUETOOTH_ENABLED;
+    }
+
+    /*
+    This method checks if the bluetooth adapter is paired with the headset device and updates the
+    bluetoothDevice instance field.
+     */
+    public BluetoothDevice returnPairedHeadset(){
+        Set<BluetoothDevice> bondedDevices = bluetoothAdapter.getBondedDevices();
+        for(BluetoothDevice device : bondedDevices)
+            if(device.getName().equals(DEVICE_NAME))
+                return device;
+        return null;
+    }
+
+    /*
+    This method creates a bluetooth connection to the headset device and returns the corresponding socket.
+    Null is returned for all other cases (non-working).
+    */
+    public BluetoothSocket createBluetoothConnection(BluetoothDevice headsetBTDevice){
+        if(headsetBTDevice != null){    // Validating once to prevent exceptions
+            bluetoothSocket = null;
+            try{
+                bluetoothSocket = headsetBTDevice.createRfcommSocketToServiceRecord(port_uuid);
+                bluetoothSocket.connect();
+            } catch (IOException e){
+                e.printStackTrace();
+                return null;
+            }
+            return bluetoothSocket;     // returning the bluetooth socket only after ensuring the connection has established
+        }
+        return null;
+    }
+
+    /*
+    This method writes the string data using the bluetoothsocket's outputstream.
+    Returns true only if it is able to successfully write the data; returns false for all the other cases.
+    */
+    public boolean sendData(String string){
+        if(bluetoothSocket != null){    // Validating once to prevent exceptions
+            OutputStream outputStream = null;
+            try{
+                outputStream = bluetoothSocket.getOutputStream();
+                outputStream.write(string.getBytes());
+            } catch (IOException e){
+                e.printStackTrace();
+                return false;
+            }
+            finally {
+                if(outputStream != null){
+                    try{
+                        outputStream.close();
+                    } catch (IOException e){
+                        e.printStackTrace();
+                    }
+                }
+            }
+            return true;
+        }
+        return false;
+    }
+
+    public void cleanUp(){
```

Comment:
We would need some reformatting to the code

Reply:
@CloudyPadmal ok would do that for sure. Thanks.

Reply:
Still it's not reformatted.